### PR TITLE
Add metric tag to differentiate long-poll and normal request

### PR DESCRIPTION
**Pull request by longquanzheng**
For https://github.com/uber/cadence-java-client/issues/649

### CODE REVIEW

File: src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java [hunk 1]
```diff
@@ -355,7 +356,14 @@ public void close() {
   }
 
   private <T> T measureRemoteCall(String scopeName, RemoteCall<T> call) throws TException {
+    return measureRemoteCallWithTags(scopeName, call, null);
+  }
+
+  private <T> T measureRemoteCallWithTags(String scopeName, RemoteCall<T> call, Map<String, String> tags) throws TException {
     Scope scope = options.getMetricsScope().subScope(scopeName);
+    if (tags != null){
```
Comment (Bostwickenator) [inline src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:364]: nit: missing space before `{`

File: src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java [hunk 2]
```diff
@@ -665,17 +673,27 @@ private StartWorkflowExecutionResponse startWorkflowExecution(
...
   }
 
   @Override
   public GetWorkflowExecutionHistoryResponse GetWorkflowExecutionHistory(
       GetWorkflowExecutionHistoryRequest request) throws TException {
-    return measureRemoteCall(
-        ServiceMethod.GET_WORKFLOW_EXECUTION_HISTORY,
-        () -> getWorkflowExecutionHistory(request, null));
+    Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, "normal");
+    if (request.isWaitForNewEvent()){
```
Comment (Bostwickenator) [inline src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:690]: nit: missing space before `{`

File: src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java [hunk 3]
```diff
@@ -665,17 +673,27 @@ private StartWorkflowExecutionResponse startWorkflowExecution(
...
+        tags);
   }
 
   @Override
   public GetWorkflowExecutionHistoryResponse GetWorkflowExecutionHistory(
       GetWorkflowExecutionHistoryRequest request) throws TException {
-    return measureRemoteCall(
-        ServiceMethod.GET_WORKFLOW_EXECUTION_HISTORY,
-        () -> getWorkflowExecutionHistory(request, null));
+    Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, "normal");
```
Comment (Bostwickenator) [inline src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:689]: I don't know uber's standards around ternary statements but I'd be tempted to do:
`Map<String, String> tags = ImmutableMap.of(MetricsTag.RequestType, (request.isWaitForNewEvent() ? "long-poll" : "normal"));`
That would avoid running `ImmutableMap.of` twice and whatever performance penalty that entails.

**Review by Bostwickenator — CHANGES_REQUESTED** [review]
Some potential improvements

File: src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java (hunk 1 above)
Comment (longquanzheng) [reply src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java:364]: Super good catch! Lol looks like I forgot to ocnfigure my autosaving plugin

**Review by mkolodezny — APPROVED** [review]

**Pull request merged**